src/RA_Dlg_Achievement.cpp

```diff
@@ -227,23 +227,25 @@ BOOL LocalValidateAchievementsBeforeCommit(int nLbxItems[1])
             return FALSE;
         }
 
-        char sIllegalChars[] = { '&', ':' };
-
-        const size_t nNumIllegalChars = sizeof(sIllegalChars) / sizeof(sIllegalChars[0]);
-        for (size_t i = 0; i < nNumIllegalChars; ++i)
+        constexpr std::array<char, 2> sIllegalChars{ '&', ':' };
+        for (auto& cNextChar : sIllegalChars)
         {
-            char cNextChar = sIllegalChars[i];
-
-            if (strchr(Ach.Title().c_str(), cNextChar) != nullptr)
+            auto ErrMsg =[&cNextChar]() noexcept
+            {
+                std::ostringstream oss;
+                oss << "Achievement title contains an illegal character: " << cNextChar << "\nPlease remove and try again";
+                return oss.str();
+            };
```

Comment:
This is incorrect. The second message was for description, not title. 

Also, I don't feel that writing a six line function (233-238) is justified given that it's only replacing two lines (what would have been 235-236 at 241 and 247). It breaks the cohesion of having the error messages tightly paired with the logic that triggers them.

If you really felt the function was necessary, the MessageBox call could have been moved into it too.

Reply:
Not a big deal. Thanks for pointing it out.